mailpoet/tests/_support/AcceptanceTester.php

```diff
@@ -677,4 +677,45 @@ public function canTestWithPlugin(string $pluginSlug): bool {
     }
     return (int)$result === 0;
   }
+
+  /**
+   * Some tests rely on background job processing.
+   * The processing runs in 1 minute interval (default Action Scheduler interval)
+   * This method triggers the processing immediately so that tests don't have to wait.
+   */
+  public function triggerMailPoetActionScheduler(): void {
+    $i = $this;
+    // Set Action Scheduler in WP Cron to run immediately
+    $wpCronOption = $i->grabOptionFromDatabase('cron');
+    if (!is_array($wpCronOption)) {
+      return;
+    }
+    $actionSchedulerRunKey = null;
+    foreach ($wpCronOption as $timestampKey => $cronRun) {
+      if (is_array($cronRun) && array_key_exists('action_scheduler_run_queue' ,$cronRun)) {
+        $actionSchedulerRunKey = $timestampKey;
+      }
+    }
+    if (!$actionSchedulerRunKey) {
+      return;
+    }
+    $actionSchedulerJob = $wpCronOption[$actionSchedulerRunKey];
+    $wpCronOption[time() - 100] = $actionSchedulerJob;
+    unset($wpCronOption[$actionSchedulerRunKey]);
+    unset($wpCronOption['version']);
+    ksort($wpCronOption);
+    $wpCronOption['version'] = 2;
+    $i->importSql([
+      "UPDATE mp_options SET option_value = '" . serialize($wpCronOption) . "' WHERE option_name = 'cron';",
+      "DELETE FROM mp_options WHERE option_name = '_transient_doing_cron';", // Remove WP Cron lock
+    ]);
+
+    // Schedule MailPoet trigger in action scheduler to run immediately
+    $i->importSql([
+      "UPDATE mp_actionscheduler_actions SET scheduled_date_gmt = SUBTIME(now(), '01:00:00'), scheduled_date_local = SUBTIME(now(), '01:00:00') WHERE hook = 'mailpoet/cron/daemon-trigger' AND status = 'pending';",
+    ]);
+
+    //Page reload triggers WP Cron and it triggers Action Scheduler
+    $i->reloadPage();
```

Comment:
Unfortunately, the admin page trigger works [also once per minute](https://github.com/woocommerce/action-scheduler/blob/2d58fd29e9a17725009d78a029b0550112437a26/docs/index.md#how-it-works).
But I guess I can try removing the lock for the action scheduler's admin page load trigger instead of rescheduling the WP Cron task. Removing the lock should be much simpler.

> And what do you think about the WP CLI approach and having WP cron and Action Scheduler disabled?
I think I tried that and ran into some issues, but I can't remember. Also, the triggering by wp-cron or the admin request is closer to what really happens.

Reply:
Ok. So I tried with the admin page trigger. It works, but in some tests (e.g. Woo purchase emails) we don't visit the admin. So I revisited the WP CLI approach. I found why it was not working, fixed that and not the code is much simpler.